### DISCUSSION TITLE
new rizin repo for same cmds for x96dbg engines dependencies

### DISCUSSION THREAD

**Discussion by gogo2464** [rizinorg/rizin#4420] [General]
**Is your feature request related to a problem? Please describe.**
I am always frustruated when I want to use rizin for windows. the philosophy of rizin is to do test and it is very good but there are too much wheel reimplementation.

**Describe the solution you'd like**
I suggest to create another repo for a fork of rizin with a different philosophy where any disassemble plugin relies on capstone, reassembles with keystone, have any x64dbg backend, etc... It must keep the same rizin commands for EACH different architectures.

**Describe alternatives you've considered**
using x96dbg

**Additional context**

**Comment by ret2libc** [comment]
What is the problem with windows? How reimplementing all plugins with keystone is going to help?

Please be specific, we can see what we can do to reduce the pain on windows.

**Reply by gogo2464** [reply]
I would like to have a soft with x96 dbg philosophy including:
- using library: recode existing modules as less a possible:
the goal is absolutely not to RECODE. it is about to avoid to recode existing libs. I would like to:
  -use capstone for disassembling
  -use titan engine to debug
It should remove a lot of incomming codebase to code. With that, x96-rizin will have any features of these engines like
  -missing tls callback on windows
  -lack of .dll/.so switching from GUI

I would like to change some x64 philo to:
-keep rizin commands
-have a vim philosophy

**Reply by gogo2464** [reply]
the windows debugger in rizin have several issues:
- no tls callback support
- no jump in dll
 a titan engine call could fit perfectly but will require deps.

I suggest to make another repo to keep rizin philosophy with no deps. The new repo willl have deps.

**Comment by gogo2464** [comment]
once again, the goal is to keep things simplest. as simple as possible. not to provide more difficult or worst more stuff to code.
